### PR TITLE
FrameGraph cleanups

### PR DESCRIPTION
**Pull request by pixelflinger**

### CODE REVIEW

File: filament/src/fg/fg/RenderTargetResource.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#include "ResourceNode.h"
+#include "VirtualResource.h"
+
+namespace filament {
+
+class FrameGraph;
+
+namespace fg {
+
+struct RenderTargetResource final : public VirtualResource {  // 104
```
Comment (prideout) [inline filament/src/fg/fg/RenderTargetResource.h]: 104?

File: filament/src/fg/fg/RenderTarget.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#include "fg/FrameGraphResource.h"
+
+#include "RenderTargetResource.h"
+
+#include <stdint.h>
+
+namespace filament {
+namespace fg {
+
+struct RenderTarget { // 32
```
Comment (prideout) [inline filament/src/fg/fg/RenderTarget.h]: This has enough logic in it such that it seems more like a class than a struct.

File: filament/src/fg/fg/RenderTargetResource.h (hunk 1 above)
Comment (prideout) [inline filament/src/fg/fg/RenderTargetResource.h]: This has enough logic in it such that it seems more like a class than a struct.

Comment (pixelflinger) [reply filament/src/fg/fg/RenderTargetResource.h]: oh yeah, it's a remainder for me of the size of this object

**Pull request merged**